components/kyma-environment-broker/internal/broker/instance_update_test.go

```diff
@@ -85,6 +88,58 @@ func TestUpdateEndpoint_UpdateSuspension(t *testing.T) {
 	assert.Len(t, response.Metadata.Labels, 1)
 }
 
+func TestUpdateEndpoint_UpdateExpirationOfTrial(t *testing.T) {
+	// given
+	instance := internal.Instance{
+		InstanceID:    instanceID,
+		ServicePlanID: TrialPlanID,
+		Parameters: internal.ProvisioningParameters{
+			PlanID: TrialPlanID,
+			ErsContext: internal.ERSContext{
+				TenantID:        "",
+				SubAccountID:    "",
+				GlobalAccountID: "",
+				Active:          nil,
+			},
+		},
+	}
+	st := storage.NewMemoryStorage()
+	st.Instances().Insert(instance)
+	st.Operations().InsertProvisioningOperation(fixProvisioningOperation("01"))
+
+	handler := &handler{}
+	q := &automock.Queue{}
+	q.On("Add", mock.AnythingOfType("string"))
+	planDefaults := func(planID string, platformProvider internal.CloudProvider, provider *internal.CloudProvider) (*gqlschema.ClusterConfigInput, error) {
+		return &gqlschema.ClusterConfigInput{}, nil
+	}
+	svc := NewUpdate(Config{}, st.Instances(), st.RuntimeStates(), st.Operations(), handler, true, false, q, planDefaults, logrus.New(), enabledDashboardConfig)
+
+	// when
+	response, err := svc.Update(context.Background(), instanceID, domain.UpdateDetails{
+		ServiceID:       "",
+		PlanID:          TrialPlanID,
+		RawParameters:   json.RawMessage(`{"expired": true}`),
+		PreviousValues:  domain.PreviousValues{},
+		RawContext:      json.RawMessage("{\"active\":false}"),
+		MaintenanceInfo: nil,
+	}, true)
+	require.NoError(t, err)
+
+	// then
+
+	assert.Equal(t, internal.ERSContext{
+		Active: ptr.Bool(false),
+	}, handler.ersContext)
+
+	require.NotNil(t, handler.Instance.Parameters.ErsContext.Active)
+	assert.True(t, *handler.Instance.Parameters.ErsContext.Active)
```

Comment:
nit: perhaps I'm reading this wrong but the update request contains 
https://github.com/kyma-project/control-plane/blob/445ba9036db31730f1073ae1f6524e606b8c0f1d/components/kyma-environment-broker/internal/broker/instance_update_test.go#L124

I would expect this to be `assert.False(..)`, no?

Reply:
the context change handler expects 2 arguments: an instance (before the change) and the new context so the handler can see how it was changed

Reply:
I know it is not clean...